### ISSUE TITLE
find/add inter-table relationships when they include a second dimension

### ISSUE THREAD

**Issue by cmgosnell** [catalyst-cooperative/pudl#2623]
Ex: in the `income_statement_ferc1` table, there is an `operating_revenues` factoid. In the `electric_operating_revenues_ferc1` there is a `electric_operating_revenues` factoid. They are conceptually related but not the same and do not have the same factoid name. The `income_statement_ferc1` table & thus the `operating_revenues` has another dimension of the `utility_type`. So there is a line of `operating_revenues` in the income statement table which is `electric`.

This one is  really easy to see bc it is `{other_dimension}_{xbrl_factoid_name}` but i expect it won't always be that straightforward.

If we add in the implied dimension in the tables where these dimensions are implied (which would probably be helpful anyway) and renamed the factoids to match we could convert these conceptual connections into more tangible ones that can be id-ed programmatically, checked and scrubbed for the 💥  

```[tasklist]
- [x] HUNT DOWN all of these inter-table connections that don't link up by name (@jrea-rmi  if you have any pointers on this that would be vv helpful)
- [x] when required, add the other dimension into the implicitly-dimension-ed table (i.e. `assign(utility_type="electric")` to `electric_operating_revenues_ferc1`.
- [ ] possibly add check into the explosion (e.g. if dimension and factoid name the same, remove duplicate)
```

**Comment by e-belfer** [comment]
Will continue to update as I go through the tables

For the electric-specific columns of `income_statement_ferc1` table:
- `operating_revenues` comes from Schedule 400 -> `electric_operating_revenues` (`electric_operating_revenues_ferc1`)
- `operating_expense` comes from Schedule 401 -> `operation_supervision` (`electric_operating_expenses_ferc1`)
- `maintenance_expense` comes from Schedule 402 -> `maintenance_supervision_and_engineering` (`electric_operating_expenses_ferc1`)
- `depreciation_expense` comes from Schedule 403 -> `depreciation_expense` in `depreciation_amortization_summary_ferc1` where `plant_function == common_plant_electric`
- `depreciation_expense_for_asset_retirement_costs` comes from Schedule 403.1 -> `depreciation_expense_asset_retirement` in `depreciation_amortization_summary_ferc1` (where `plant_function == common_plant_electric` only??)

- `amortization_and_depletion_of_utility_plant` comes from Schedule 404-5 -> ??
- `amortization_of_electric_plant_acquisition_adjustments` comes from Schedule 406 -> ??
- `amortization_of_property_losses_unrecovered_plant_and_regulatory_study_costs` comes from Schedule 407 -> ??
- `amortization_of_conversion_expenses` comes from Schedule 407 -> ??
- `regulatory_debits` comes from Schedule 407.3 -> ??
- `regulatory_credits` comes from Schedule 407.4 -> ??
-

**Comment by e-belfer** [comment]
@jrea-rmi Just circling back to this. Do you all have a working list of fields which are not identically named but are effectively identical and should be compared across tables? See above for some examples. We're particularly looking at 'second dimension' relationships (e.g. electric plants), but if there are others you're aware of this is also helpful to know!

**Comment by jrea-rmi** [comment]
@e-belfer I don't already have a compiled list like this, but I could review my previous table connections and see if I have additions to what you have.

A few balance sheet fields to add to your income statement notes are:
- regulatory assets (from assets side of balance sheet) is a detail table
- regulatory liabilities (from liabilities side of balance sheet) is a detail table
- several fields from utility plant section of balance sheet assets should match summary of utility plant fields
- electric plant in service and accumulated provision for depreciation, amortization... tables are components of the summary of utility plant table

**Comment by e-belfer** [comment]
Thanks for the follow-up and the suggestions! I'll work on compiling a list and push it to a PR for you to look over to see that nothing's missing or out-of-place.

**Comment by cmgosnell** [comment]
closing this issue bc we merged #2669 into `explode_ferc1`
